warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java

```diff
@@ -68,74 +68,126 @@
 import org.apache.log4j.Logger;
 
 import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.NoSuchElementException;
 import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
 
 /**
  * Determine if a node can be processed against the global index and/or the field index
- * 
+ * <p>
  * A node can be processed against the field index if it is not a ER/NR/LE/LT/GE/GT node (unless surrounded by an ivarator expression).
- * 
+ * <p>
  * A node can be processed against the global index if it is not a ER,NR,LE,LT,GE,GT,NOT, or NE (unless surrounded by an ivarator expression).
- * 
+ * <p>
  * In general an OR can be processed if it is completely composed of expressions that can be processed. An AND can be processed it at least one of its children
  * can be processed.
- * 
  */
 public class ExecutableDeterminationVisitor extends BaseVisitor {
     
     /**
-     * EXECUTABLE means that the expression is executable against the index PARTIAL means that we have an OR that cannot be completely satisfied by the index
-     * NON_EXECUTABLE means that the expression cannot be executed against the index IGNORABLE means that it does not matter the executable state of the
-     * underlying expression ERROR means that we have an expression that is index only but yet cannot be run against the index (negation, delayed prefix)
+     * Represents the executability status of an expression against the index.
      */
     public enum STATE {
-        EXECUTABLE, PARTIAL, NON_EXECUTABLE, IGNORABLE, ERROR
-    }
-    
-    private static final Logger log = Logger.getLogger(ExecutableDeterminationVisitor.class);
-    
-    private interface Output {
-        void writeLine(String line);
+        
+        /**
+         * The expression is executable against the index.
+         */
+        EXECUTABLE(2, 3, 3),
+        
+        /**
+         * The expression has an OR or cannot be completely satisfied by the index.
+         */
+        PARTIAL(1, 1, 2),
+        
+        /**
+         * The expression cannot be executed against the index.
+         */
+        NON_EXECUTABLE(3, 2, 1),
+        
+        /**
+         * The expression does not matter for determining executability against the global index.
+         */
+        IGNORABLE(4, 4, 4),
+        
+        /**
+         * The expression is index-only, but cannot be run against the index (negation, delayed prefix).
+         */
+        ERROR(0, 0, 0);
```

Comment:
I see what you were trying to do here for efficiency, but this impacts clarity of the code quite a bit. In this case I think the clarity/maintainability is more important than the efficiency gain.

Reply:
I re-implemented this in a different way that removes the need to store the priority ordering in the `STATE` enum. Let me know if you think it requires further changes.